src/main/java/com/ivanmostovyi/demo/service/MessageServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.ivanmostovyi.demo.service;
+
+import com.ivanmostovyi.demo.domain.Messages;
+import com.ivanmostovyi.demo.domain.User;
+import com.ivanmostovyi.demo.dto.MessagesFormDto;
+import com.ivanmostovyi.demo.repository.MessagesRepository;
+import com.ivanmostovyi.demo.repository.UserRepository;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MessageServiceImpl implements MessageService {
+
+    private UserRepository userRepository;
+
+    private MessagesRepository messagesRepository;
+
+    public MessageServiceImpl(UserRepository userRepository, MessagesRepository messagesRepository) {
+        this.userRepository = userRepository;
+        this.messagesRepository = messagesRepository;
+    }
+
+    @Override
+    public void create(MessagesFormDto messagesFormDto){
+
+        Optional<User> receiverUser = userRepository.findByUsername(messagesFormDto.getReceiverUsername());
+
+        receiverUser.orElseThrow(() -> new UsernameNotFoundException("Receiver not found!"));
```

Comment:
Do not reuse security exception.

Just create your own: `ServiceException` for general use or `MessageSendingException` for this specific case, depending on your desing and how would you like to handle this case